plugins/modules/grafana_silence.py

```diff
@@ -0,0 +1,369 @@
+            ):
+                return response
+        return None
+
+    def get_silence_by_id(self, silence_id):
+        url = "/api/alertmanager/grafana/api/v2/silence/{SilenceId}".format(
+            SilenceId=silence_id
+        )
+        response = self._send_request(url, headers=self.headers, method="GET")
+        return response
+
+    def get_silences(self):
+        url = "/api/alertmanager/grafana/api/v2/silences"
+        response = self._send_request(url, headers=self.headers, method="GET")
+        return response
+
+    def delete_silence(self, silence_id):
+        url = "/api/alertmanager/grafana/api/v2/silence/{SilenceId}".format(
+            SilenceId=silence_id
+        )
+        response = self._send_request(url, headers=self.headers, method="DELETE")
+        return response
+
+
+def setup_module_object():
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+        required_together=base.grafana_required_together(),
+        mutually_exclusive=base.grafana_mutually_exclusive(),
+    )
+    return module
+
+
+argument_spec = base.grafana_argument_spec()
+argument_spec.update(
+    comment=dict(type="str", required=True),
+    state=dict(type="str", choices=["present", "absent"], default="present"),
+    created_by=dict(type="str", required=True),
+    starts_at=dict(type="str", required=True),
+    ends_at=dict(type="str", required=True),
+    matchers=dict(type="list", elements="dict", required=True),
+    skip_version_check=dict(type="bool", default=False),
+)
+
+
+def main():
+
+    module = setup_module_object()
+    comment = module.params["comment"]
+    created_by = module.params["created_by"]
+    starts_at = module.params["starts_at"]
+    ends_at = module.params["ends_at"]
+    matchers = module.params["matchers"]
+    state = module.params["state"]
+
+    changed = False
+    failed = False
+    grafana_iface = GrafanaSilenceInterface(module)
+
+    silence = grafana_iface.get_silence(
+        comment, created_by, starts_at, ends_at, matchers
+    )
+
+    if state == "present":
+
+        if not silence:
+            silence = grafana_iface.create_silence(
+                comment, created_by, starts_at, ends_at, matchers
+            )
+            silence = grafana_iface.get_silence_by_id(silence["silenceID"])
+            changed = True
+        else:
+            module.exit_json(
+                failed=failed,
+                changed=changed,
+                msg="Silence with same parameters already exists! eg. '%s'"
+                % silence["id"],
+            )
+    elif state == "absent":
+        if silence:
+            grafana_iface.delete_silence(silence["id"])
+            changed = True
+        else:
+            module.exit_json(
+                failed=False,
+                changed=changed,
+                msg="No silence with the same parameters found! Deletion not possible",
```

Comment:
To keep it in sync with the other modules, let's change this to:
```suggestion
                msg="Silence deleted.",
```